spec/models/results/response_tree_builder_spec.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+describe "response tree" do
+  # form with repeat group
+  context "simple form" do
+    let(:form1) { create(:form, question_types: %w[text text text]) }
+
+    it "should produce a simple tree from a form with three children" do
+      response_tree = Results::ResponseTreeBuilder.new(form1).build
+      # same number of leaves at the root
+      expect(response_tree.root_answers.count).to eq form1.root_questionings.count
+
+      # expect to be the same qing_ids
+      expect(response_tree.root_answers.map(&:questioning_id))
+        .to include(form1.root_questionings[0].id, form1.root_questionings[1].id,
+          form1.root_questionings[2].id)
```

Comment:
You should actually just do `expect(response_tree.children.map(&:questioning_id)).to eq(form_tree.children.map(&:id))` b/c you know the order should match also.